### ISSUE TITLE
ZRANK doesn't work as expected like in redis

### ISSUE THREAD

**Issue by pyros2097** [yinqiwen/ardb#168]
In ardb
```
 zrange my_set 0 -1
 1) "user4"
 2) "11"
 3) "user6"
 4) "11"
 5) "user3"
 6) "15"
 7) "user1"
 8) "30"
 9) "user2"
10) "30"
11) "user5"
12) "122"

zrevrank my_set user6
5
zrevrank my_set user4
5
```
whereas in redis
```
zrange my_set 0 -1
 1) "user4"
 2) "11"
 3) "user6"
 4) "11"
 5) "user3"
 6) "15"
 7) "user1"
 8) "30"
 9) "user2"
10) "30"
11) "user5"
12) "122"

zrevrank my_set user6
4
zrevrank my_set user4
5
```
using rocksdb as backend
I think this could be because of the ordering of keys in the db? Is this the way this should work?

I think they call this a league ranking or a normal leaderboard ranking which is what we actually implemented over redis using zrevrangebyscore but ardb has this inbuilt that's great!!

**Comment by yinqiwen** [comment]
Current `rank` operations in ardb just  rank scores,  if two value have same score, they would  have same rank. 
This is a bug which would be fixed later.

**Comment by gonzajg** [comment]
Hi, I'm having the same problem with elements with the same score. Redis uses lexigraphical order in this case and ARDB uses other somehow erratic (maybe depending on the order of insertion). So my algorithms based on that order are broken.

Is there any fix in the pipeline? 

Thanks!

**Comment by yinqiwen** [comment]
fixed in 0.9
